### PR TITLE
1 hashmap

### PR DESCRIPTION
**Pull request by varchenkomichael**

### CODE REVIEW

File: src/main/java/com/varchenko/MikeHashMap.java [hunk 1]
```diff
@@ -1,7 +1,188 @@
...
+        }
+        return buckets[indexFor(key)].getValue();
+    }
+
+    @Override
+    public V remove(Object key) {
+        Node<K, V> node = buckets[indexFor(key)];
+        if (node != null) {
+            if (node.getKey().hashCode() == key.hashCode() && node.getKey().equals(key)) {
+                findInNode(node, key).setValue(null);
```
Comment (bryzhatov) [inline src/main/java/com/varchenko/MikeHashMap.java:105]: Error

File: src/main/java/com/varchenko/MikeHashMap.java [hunk 2]
```diff
@@ -1,7 +1,188 @@
...
+        this();
+        this.capacity = capacity;
+    }
+
+    private void resize() {
+        if (threshold >= capacity * LOAD_FACTOR) {
+            int bucketSize = capacity * 2;
+            Node<K, V>[] oldBucket = buckets;
+            buckets = new Node[bucketSize];
+            currentSize = 0;
```
Comment (bryzhatov) [inline src/main/java/com/varchenko/MikeHashMap.java]: Why

File: src/main/java/com/varchenko/MikeHashMap.java [hunk 3]
```diff
@@ -1,7 +1,188 @@
...
+            for (Node<K, V> kvNode : oldBucket) {
+                if (kvNode != null)
+                    put(kvNode.getKey(), kvNode.getValue());
+                currentSize++;
```
Comment (bryzhatov) [inline src/main/java/com/varchenko/MikeHashMap.java]: why

File: src/main/java/com/varchenko/MikeHashMap.java [hunk 4]
```diff
@@ -1,7 +1,205 @@
...
+        this();
+        this.capacity = capacity;
+    }
+
+    // todo issue )
+    private void resize() {
+        if (size >= capacity * LOAD_FACTOR) {
+            int bucketSize = capacity + (capacity >> 1);
+            Node<K, V>[] oldBucket = buckets;
+            buckets = new Node[bucketSize];
```
Comment (bryzhatov) [inline src/main/java/com/varchenko/MikeHashMap.java]: Issue / refactor

File: src/main/java/com/varchenko/MikeHashMap.java [hunk 5]
```diff
@@ -1,7 +1,205 @@
...
+                }
+            }
+            return findInNode(buckets[indexFor(k)], k).getValue();
+        }
+        return null;
+    }
+
+    // todo issue )
+    @Override
+    public V put(K key, V value) {
```
Comment (bryzhatov) [inline src/main/java/com/varchenko/MikeHashMap.java:86]: please, refactor this code

File: src/test/java/com/varchenko/MikeHashMapTest.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        mikeHashMap.put("get6", "test2");
+        mikeHashMap.put("get5", "test3");
+        mikeHashMap.put("get4", "test4");
+        mikeHashMap.put("get3", "test5");
+        mikeHashMap.put("get2", "test7");
+        mikeHashMap.put("get1", "test8");
+        assertEquals(9, mikeHashMap.size());
+    }
+
+}
```
Comment (bryzhatov) [inline src/test/java/com/varchenko/MikeHashMapTest.java:80]: remove .class files below and use git ignore correctly

File: src/main/java/com/varchenko/MikeHashMap.java [hunk 7]
```diff
@@ -1,7 +1,205 @@
...
+        };
+    }
+
+    @Override
+    public int size() {
+        return size;
+    }
+
+    @Override
+    public String toString() {
```
Comment (bryzhatov) [inline src/main/java/com/varchenko/MikeHashMap.java:162]:
```
Use this format for toString 
[1]
- Entry 1
- Entry 2
[2]
- Entry 3
- Entry 4
```

File: src/main/java/com/varchenko/Main.java [hunk 8]
```diff
@@ -0,0 +1,12 @@
+package com.varchenko;
+
+import java.util.Map;
+
+public class Main {
```
Comment (bryzhatov) [inline src/main/java/com/varchenko/Main.java]: Remove this class

File: src/main/java/com/varchenko/MikeHashMap.java [hunk 9]
```diff
@@ -1,7 +1,205 @@
...
+    private static final int DEFAULT_CAPACITY = 16;
+    private Node<K, V>[] buckets;
+    private Node<K, V> current;
+    private int capacity;
+    private int size;
+
+
+    public MikeHashMap() {
+        this.capacity = DEFAULT_CAPACITY;
+        buckets = new Node[capacity];
```
Comment (bryzhatov) [inline src/main/java/com/varchenko/MikeHashMap.java:16]: Refactor it

File: src/main/java/com/varchenko/MikeHashMap.java (hunk 4 above)
Comment (varchenkomichael) [reply src/main/java/com/varchenko/MikeHashMap.java]: done

File: src/main/java/com/varchenko/Main.java (hunk 8 above)
Comment (varchenkomichael) [reply src/main/java/com/varchenko/Main.java]: done

File: src/main/java/com/varchenko/MikeHashMap.java (hunk 9 above)
Comment (varchenkomichael) [reply src/main/java/com/varchenko/MikeHashMap.java:16]: done

File: src/main/java/com/varchenko/MikeHashMap.java (hunk 7 above)
Comment (varchenkomichael) [reply src/main/java/com/varchenko/MikeHashMap.java:162]: (key: ; value: )

**Pull request closed without merging**